runtime/common/src/migrations/restricted_location.rs

```diff
@@ -0,0 +1,141 @@
+use cfg_primitives::AccountId;
+use cfg_types::{locations::RestrictedTransferLocation, tokens::FilterCurrency};
+use frame_support::{
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+};
+use pallet_transfer_allowlist::AccountCurrencyTransferAllowance;
+use parity_scale_codec::Encode;
+use sp_core::H256;
+use sp_runtime::traits::{BlakeTwo256, Hash};
+use sp_std::vec::Vec;
+use staging_xcm::v4;
+
+mod old {
+	use cfg_primitives::AccountId;
+	use cfg_types::{domain_address::DomainAddress, tokens::FilterCurrency};
+	use frame_support::{pallet_prelude::*, storage_alias};
+	use frame_system::pallet_prelude::*;
+	use hex::FromHex;
+	use pallet_transfer_allowlist::AllowanceDetails;
+	use sp_core::H256;
+	use staging_xcm::v3;
+
+	#[derive(
+		Clone, RuntimeDebugNoBound, Encode, Decode, Eq, PartialEq, MaxEncodedLen, TypeInfo,
+	)]
+	pub enum RestrictedTransferLocation {
+		Local(AccountId),
+		Xcm(H256),
+		Address(DomainAddress),
+	}
+
+	#[storage_alias]
+	pub type AccountCurrencyTransferAllowance<T: pallet_transfer_allowlist::Config> = StorageNMap<
+		pallet_transfer_allowlist::Pallet<T>,
+		(
+			NMapKey<Twox64Concat, AccountId>,
+			NMapKey<Twox64Concat, FilterCurrency>,
+			NMapKey<Blake2_128Concat, RestrictedTransferLocation>,
+		),
+		AllowanceDetails<BlockNumberFor<T>>,
+		OptionQuery,
+	>;
+
+	pub fn location_v3_created_by_apps(_account_id: &AccountId) -> v3::Location {
+		// Ref: https://github.com/centrifuge/apps/blob/b59bdd34561a4ccd90e0d803c14a3729fc2f3a6d/centrifuge-app/src/utils/usePermissions.tsx#L386
+		// for account_id == "4dTeMxuPJCK7zQGhFcgCivSJqBs9Wo2SuMSQeYCCuVJ9xrE2"
+		v3::Location::new(
+			1,
+			v3::Junctions::X2(
+				v3::Junction::Parachain(1000), // AssetHub
+				v3::Junction::AccountId32 {
+					network: None,
+					id: <[u8; 32]>::from_hex(
+						// Address used by Anemoy to withdraw in AssetHub
+						"10c03288a534d77418e3c19e745dfbc952423e179e1e3baa89e287092fc7802f",
+					)
+					.expect("keep in the array"),
+				},
+			),
+		)
+	}
+}
```

Comment:
Right now, there are two accounts in Demo. I will add a migration for those based on your branch.

```
["kAHTP9K77aYEKgn7gNLjiNogVPPJRoEy8QBmEmfbQndBywEC5",{"Specific":{"ForeignAsset":"100,001"}},{"XCM":"0x4607f5b72826a88cc19c87a35a1f901388b6980ee98edd3d45a4cf1a26485b40"}]
{"allowedAt":"0","blockedAt":"1,816,931"}

["kAHTP9K77aYEKgn7gNLjiNogVPPJRoEy8QBmEmfbQndBywEC5",{"Specific":{"ForeignAsset":"100,001"}},{"XCM":"0x992896d5bfc1ddd6f58023d74bf318c3119cc0b2db2c946b07123aa0893834dc"}]
{"allowedAt":"0","blockedAt":"4,294,967,295"}

```

I was wondering whether want to provide a migration extrinsic which requires an `AccountId` input and updates the location based on our assumption of the pre-state boilerplate from Apps.

Reply:
> I was wondering whether want to provide a migration extrinsic which requires an AccountId input and updates the location based on our assumption of the pre-state boilerplate from Apps.


Mmm... interesting. That would be great in case new entries appear until RU, because we basically need to hardcode them. Could make sense then to remove the migration and add the "migration extrinsic"